cmd/edit/machinepool/machinepool.go

```diff
@@ -26,43 +26,62 @@ func editMachinePool(cmd *cobra.Command, machinePoolID string, clusterKey string
 		os.Exit(1)
 	}
 
+	isMinReplicasSet := cmd.Flags().Changed("min-replicas")
+	isMaxReplicasSet := cmd.Flags().Changed("max-replicas")
+	isReplicasSet := cmd.Flags().Changed("replicas")
+	isAutoscalingSet := cmd.Flags().Changed("enable-autoscaling")
+	isLabelsSet := cmd.Flags().Changed("labels")
+	isTaintsSet := cmd.Flags().Changed("taints")
+
 	// Editing the default machine pool is a different process
 	if machinePoolID == "Default" {
-		if cmd.Flags().Changed("labels") {
-			r.Reporter.Errorf("Labels cannot be updated on the Default machine pool")
-			os.Exit(1)
-		}
-		if cmd.Flags().Changed("taints") {
+		if isTaintsSet {
 			r.Reporter.Errorf("Taints are not supported on the Default machine pool")
 			os.Exit(1)
 		}
 
-		autoscaling, replicas, minReplicas, maxReplicas := getMachinePoolReplicas(cmd, r.Reporter, machinePoolID,
-			cluster.Nodes().Compute(), cluster.Nodes().AutoscaleCompute())
+		// if no value set enter interactive mode
+		if !(isMinReplicasSet || isMaxReplicasSet || isReplicasSet || isAutoscalingSet || isLabelsSet) {
+			interactive.Enable()
+		}
+
+		clusterConfig := ocm.Spec{}
+
+		autoscaling, replicas, minReplicas, maxReplicas, scalingUpdated :=
+			getMachinePoolReplicas(cmd, r.Reporter, machinePoolID, cluster.Nodes().Compute(),
+				cluster.Nodes().AutoscaleCompute(), !isLabelsSet)
+
+		if scalingUpdated {
+			if cluster.MultiAZ() {
+				if !autoscaling && replicas < 3 ||
+					(autoscaling && isMinReplicasSet && minReplicas < 3) {
+					r.Reporter.Errorf("Default machine pool for AZ cluster requires at least 3 compute nodes")
+					os.Exit(1)
+				}
 
-		if cluster.MultiAZ() {
-			if !autoscaling && replicas < 3 ||
-				(autoscaling && cmd.Flags().Changed("min-replicas") && minReplicas < 3) {
-				r.Reporter.Errorf("Default machine pool for AZ cluster requires at least 3 compute nodes")
+				if !autoscaling && replicas%3 != 0 ||
```

Comment:
I think this condition applies only to multi-AZ clusters.

Reply:
That is why it's under the `if` on line 55